### PR TITLE
Fix script resolution

### LINKED ISSUE #104: Module scripts src relative path not prefixed by `.` is not correctly rewritten during build
For

https://github.com/snowpackjs/astro/blob/d2337e70239137700447dd46d5bd70de38a0a8ae/examples/snowpack/astro/components/Nav.astro#L351

If you replace it with `<script type="module" src="docsearch.js"></script>`, the source path would not be rewritten and kept as `docsearch.js`, thus making the output page not able to load this script.

This is due to

https://github.com/snowpackjs/astro/blob/d2337e70239137700447dd46d5bd70de38a0a8ae/src/compiler/transform/module-scripts.ts#L29

and thus would miss relative paths without `.` prefix and not performing the rewrite (technically I believe only `/path-absolute` paths should be ignored). To fix this, probably we can do

```ts
if (!src || src.startsWith('/')) {
```

(I had a local branch for this that seems working, but since the project could not be forked I could not make the PR I'll just leave my investigations here)

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

Fixes #104.

## Testing

Adds tests

## Docs

No docs needed

### CODE REVIEW

File: packages/astro/src/build.ts [hunk 1]
```diff
@@ -110,6 +110,9 @@ export async function build(astroConfig: AstroConfig, logging: LogOptions = defa
           scanPromises.push(
             runtime.load(url).then((result) => {
               if (result.statusCode !== 200) {
+                if (result.statusCode === 404) {
+                  throw new Error(`${buildState[id].srcPath.href}: could not find "${path.basename(url)}"`);
```
Comment (drwpow) [inline packages/astro/src/build.ts:114]: @matthewp when I try and display a helpful error message here, this is what I end up seeing:

Screen Shot 2021-06-14 at 16 22 29

It has something to do with how this is being passed to the logger, but I couldn’t figure this out. How can we fix this?

(to recreate this, simply try and run `astro build` with an asset that doesn’t exist, such as `<script src="foo.js">`)

File: packages/astro/test/fixtures/astro-resolve/src/pages/scripts.astro [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <head></head>
+
+  <body>
+    <!-- not `type="module"` -->
+    <script src="./relative.js"></script>
+
+    <!-- no `src` -->
+    <script></script>
+
+    <!-- type="module" but no leading "." -->
```
Comment (FredKSchott) [inline packages/astro/test/fixtures/astro-resolve/src/pages/scripts.astro:11]: shouldn't we add a test for:

```
    <!-- type="module" but leading "." -->
    <script type="module" src="./relative.js"></script>
```

**Review by FredKSchott — CHANGES_REQUESTED** [review]
Can we talk about this one as well before merging? It feels odd to resolve some attributes but not others, and I had assumed we were taking a stance of: "whatever HTML you give us, is what we send to the page". Sounds like maybe I missed a Discord chat?

Both this and #437 just feel like a really big step to be taking and I just want to make sure that we've thought this through before going down this path.

**Comment by drwpow** [comment]
> Can we talk about this one as well before merging? It feels odd to resolve some attributes but not others, and I had assumed we were taking a stance of: "whatever HTML you give us, is what we send to the page". Sounds like maybe I missed a Discord chat?
> 
> Both this and #437 just feel like a really big step to be taking and I just want to make sure that we've thought this through before going down this path.

Like #437 this also isn’t a change in behavior; we’ve done this for a while. This just fixes a bug where we unintentionally transformed paths that had a `.` but didn’t if they ommitted that (the original issue). Sure, we can talk about changing this for sure, but we do need to fix the bugs first, and both these PRs are only bugfixes.

**Review by FredKSchott — DISMISSED** [review]

**Pull request merged**